lib/utils/utils.go

```diff
@@ -203,6 +203,18 @@ func SliceContainsStr(slice []string, value string) bool {
 	return false
 }
 
+// ParseCompatibility parses the compatibility parameter and makes sure it's
+// valid.
+func ParseCompatibility(s string) (string, error) {
+	if s == teleport.CompatibilityNone {
```

Comment:
this is a really long/complicated way to write a function like this. it is also misleading, since it does not do any parsing. a much simpler way to express the same thing would look like

```
switch s {
     case a, b: return s, nil
     default: return error.
}